spec/project_spec.rb

```diff
@@ -1018,21 +1099,34 @@ class InvalidErrorHandler; end
     it 'should return nil if feature_flag_key is nil' do
       expect(project_instance.get_feature_variable_integer(nil, 'integer_variable', user_id, user_attributes))
         .to eq(nil)
-      expect(spy_logger).to have_received(:log).once.with(Logger::ERROR, 'Feature flag key cannot be empty.')
+      expect(spy_logger).to have_received(:log).once.with(Logger::ERROR, 'Feature flag key is invalid')
     end
 
     it 'should return nil if variable_key is nil' do
       expect(project_instance.get_feature_variable_integer('integer_single_variable_feature', nil, user_id, user_attributes))
         .to eq(nil)
-      expect(spy_logger).to have_received(:log).once.with(Logger::ERROR, 'Variable key cannot be empty.')
+      expect(spy_logger).to have_received(:log).once.with(Logger::ERROR, 'Variable Key is invalid')
     end
 
     it 'should return nil if user_id is empty or nil' do
       expect(project_instance.get_feature_variable_integer('integer_single_variable_feature', 'integer_variable', nil, user_attributes))
         .to eq(nil)
       expect(project_instance.get_feature_variable_integer('integer_single_variable_feature', 'integer_variable', '', user_attributes))
         .to eq(nil)
-      expect(spy_logger).to have_received(:log).twice.with(Logger::ERROR, 'User ID cannot be empty.')
+      expect(spy_logger).to have_received(:log).twice.with(Logger::ERROR, 'User ID is invalid')
+    end
+
+    it 'should call inputs_valid? with the proper arguments in get_feature_variable_integer' do
```

Comment:
Change this to get_feature_variable_xxx method. Because you have called input_validate in getFeatureVariableValueForType. So your intent is to assert that it will get called in any of the accessor methods. Not just feature variable integer. If I am right